converter/internal/otelcolconvert/converter_otlpreceiver.go

```diff
@@ -0,0 +1,221 @@
+		WriteBufferSize:      units.Base2Bytes(cfg.WriteBufferSize),
+
+		Keepalive: toKeepaliveServerArguments(cfg.Keepalive),
+
+		IncludeMetadata: cfg.IncludeMetadata,
+	}
+}
+
+func toTLSServerArguments(cfg *configtls.TLSServerSetting) *otelcol.TLSServerArguments {
+	if cfg == nil {
+		return nil
+	}
+
+	return &otelcol.TLSServerArguments{
+		TLSSetting: toTLSSetting(cfg.TLSSetting),
+
+		ClientCAFile: cfg.ClientCAFile,
+	}
+}
+
+func toTLSSetting(cfg configtls.TLSSetting) otelcol.TLSSetting {
+	return otelcol.TLSSetting{
+		CA:             string(cfg.CAPem),
+		CAFile:         cfg.CAFile,
+		Cert:           string(cfg.CertPem),
+		CertFile:       cfg.CertFile,
+		Key:            rivertypes.Secret(cfg.KeyPem),
+		KeyFile:        cfg.KeyFile,
+		MinVersion:     cfg.MinVersion,
+		MaxVersion:     cfg.MaxVersion,
+		ReloadInterval: cfg.ReloadInterval,
+	}
+}
+
+func toKeepaliveServerArguments(cfg *configgrpc.KeepaliveServerConfig) *otelcol.KeepaliveServerArguments {
+	if cfg == nil {
+		return nil
+	}
+
+	return &otelcol.KeepaliveServerArguments{
+		ServerParameters:  toKeepaliveServerParameters(cfg.ServerParameters),
+		EnforcementPolicy: toKeepaliveEnforcementPolicy(cfg.EnforcementPolicy),
+	}
+}
+
+func toKeepaliveServerParameters(cfg *configgrpc.KeepaliveServerParameters) *otelcol.KeepaliveServerParamaters {
+	if cfg == nil {
+		return nil
+	}
+
+	return &otelcol.KeepaliveServerParamaters{
+		MaxConnectionIdle:     cfg.MaxConnectionIdle,
+		MaxConnectionAge:      cfg.MaxConnectionAge,
+		MaxConnectionAgeGrace: cfg.MaxConnectionAgeGrace,
+		Time:                  cfg.Time,
+		Timeout:               cfg.Timeout,
+	}
+}
+
+func toKeepaliveEnforcementPolicy(cfg *configgrpc.KeepaliveEnforcementPolicy) *otelcol.KeepaliveEnforcementPolicy {
+	if cfg == nil {
+		return nil
+	}
+
+	return &otelcol.KeepaliveEnforcementPolicy{
+		MinTime:             cfg.MinTime,
+		PermitWithoutStream: cfg.PermitWithoutStream,
+	}
+}
+
+func toHTTPConfigArguments(cfg *otlpreceiver.HTTPConfig) *otlp.HTTPConfigArguments {
+	if cfg == nil {
+		return nil
+	}
+
+	return &otlp.HTTPConfigArguments{
+		HTTPServerArguments: toHTTPServerArguments(cfg.HTTPServerSettings),
+
+		TracesURLPath:  cfg.TracesURLPath,
+		MetricsURLPath: cfg.MetricsURLPath,
+		LogsURLPath:    cfg.LogsURLPath,
+	}
+}
+
+func toHTTPServerArguments(cfg *confighttp.HTTPServerSettings) *otelcol.HTTPServerArguments {
+	if cfg == nil {
+		return nil
+	}
+
+	return &otelcol.HTTPServerArguments{
+		Endpoint: cfg.Endpoint,
+
+		TLS: toTLSServerArguments(cfg.TLSSetting),
+
+		CORS: toCORSArguments(cfg.CORS),
+
+		MaxRequestBodySize: units.Base2Bytes(cfg.MaxRequestBodySize),
+		IncludeMetadata:    cfg.IncludeMetadata,
+	}
+}
+
+func toCORSArguments(cfg *confighttp.CORSSettings) *otelcol.CORSArguments {
+	if cfg == nil {
+		return nil
+	}
+
+	return &otelcol.CORSArguments{
+		AllowedOrigins: cfg.AllowedOrigins,
+		AllowedHeaders: cfg.AllowedHeaders,
+
+		MaxAge: cfg.MaxAge,
+	}
+}
+
+type tokenizedConsumer struct {
```

Comment:
I don't love seeing new types buried this deep into a file. WDYT about making a new file for these tokenization types for otel?

Reply:
Moved to an converter_helpers.go file